core/arch/arm/plat-zynqmp/sta_bsload.c

```
@@ -0,0 +1,366 @@
+{
+	if (!csu_base) {
+		csu_base = (vaddr_t)phys_to_virt(CSU_PA_BASE, MEM_AREA_IO_SEC);
+		if (!csu_base)
+			return TEE_ERROR_ITEM_NOT_FOUND;
+	}
+
+	if (!csudma_base) {
+		csudma_base = (vaddr_t)phys_to_virt(CSUDMA_PA_BASE,
+						    MEM_AREA_IO_SEC);
+		if (!csudma_base)
+			return TEE_ERROR_ITEM_NOT_FOUND;
+	}
+
+	return TEE_SUCCESS;
+}
+
+/*
+ * Called when the instance of the TA is destroyed if the TA has not
+ * crashed or panicked. This is the last call in the TA.
+ */
+void TA_DestroyEntryPoint(void)
+{
+}
+
+/*
+ * Called when a new session is opened to the TA. *sess_ctx can be updated
+ * with a value to be able to identify this session in subsequent calls to the
+ * TA.
+ */
+TEE_Result TA_OpenSessionEntryPoint(uint32_t param_types,
+		TEE_Param  params[4], void **sess_ctx)
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+	if (param_types != exp_param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	/* Unused parameters */
+	(void)&params;
+	(void)&sess_ctx;
+
+	return TEE_SUCCESS;
+}
+
+/*
+ * Called when a session is closed, sess_ctx hold the value that was
+ * assigned by TA_OpenSessionEntryPoint().
+ */
+void TA_CloseSessionEntryPoint(void *sess_ctx)
+{
+	(void)&sess_ctx; /* Unused parameter */
+}
+
+/**
+ * load_bistream - Load a bitstream into the PL
+ * @params[0]: Memref describing buffer holding the bitstream
+ * @params[1]: Value holding flags.
+ */
+static TEE_Result load_bitstream(uint32_t param_types, TEE_Param params[4])
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+						   TEE_PARAM_TYPE_VALUE_INPUT,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (param_types != exp_param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	DMSG("bitstream va:%p, size:%#x\n",
+			params[0].memref.buffer, params[0].memref.size);
+	TEE_ASSERT(params[0].memref.buffer && params[0].memref.size);
```

Better to return an error here than panic the system.